routing.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/ipfs/go-ipns"
+	ipns_pb "github.com/ipfs/go-ipns/pb"
+	ic "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+)
+
+type proxyRouting struct {
+	kuboRPC    []string
+	httpClient *http.Client
+	rand       *rand.Rand
+}
+
+func newProxyRouting(kuboRPC []string, client *http.Client) routing.ValueStore {
+	if client == nil {
+		client = http.DefaultClient
+	}
+
+	s := rand.NewSource(time.Now().Unix())
+	rand := rand.New(s)
+
+	return &proxyRouting{
+		kuboRPC:    kuboRPC,
+		httpClient: client,
+		rand:       rand,
+	}
+}
+
+func (ps *proxyRouting) PutValue(context.Context, string, []byte, ...routing.Option) error {
+	return routing.ErrNotSupported
+}
+
+func (ps *proxyRouting) GetValue(ctx context.Context, k string, opts ...routing.Option) ([]byte, error) {
+	return ps.fetch(ctx, k)
+}
+
+func (ps *proxyRouting) SearchValue(ctx context.Context, k string, opts ...routing.Option) (<-chan []byte, error) {
+	if !strings.HasPrefix(k, "/ipns/") {
+		return nil, routing.ErrNotSupported
+	}
+
+	ch := make(chan []byte)
+
+	go func() {
+		v, err := ps.fetch(ctx, k)
+		if err != nil {
+			close(ch)
+		} else {
+			ch <- v
+			close(ch)
+		}
+	}()
+
+	return ch, nil
+}
+
+func (ps *proxyRouting) fetch(ctx context.Context, key string) ([]byte, error) {
+	key = strings.TrimPrefix(key, "/ipns/")
+	id, err := peer.IDFromBytes([]byte(key))
+	if err != nil {
+		return nil, err
+	}
+
+	key = "/ipns/" + peer.ToCid(id).String()
+
+	// Naively choose one of the Kubo RPC clients.
+	endpoint := ps.kuboRPC[rand.Intn(len(ps.kuboRPC))]
+
+	u, err := url.Parse(fmt.Sprintf("%s/api/v0/routing/get?arg=%s", endpoint, key))
```

Comment:
:information_source: exposing this is tracked in https://github.com/protocol/bifrost-infra/issues/2327